packages/jaeger-ui/src/utils/tracking/ga.tsx

```diff
@@ -24,6 +23,40 @@ import { logTrackingCalls } from './utils';
 import { getAppEnvironment, shouldDebugGoogleAnalytics } from '../constants';
 import parseQuery from '../parseQuery';
 
+// Modify the global scope to add the `gtag` function and the `dataLayer` array
```

Comment:
Why do we need to attach gtag function to the window object? Could we not use the same approach as in ReactGA where it had some module-level globals?

Reply:
based on https://developers.google.com/tag-platform/gtagjs/install, I think only the dataLayer needs to be public, the gtag function can be private in this module

Reply:
Ahh yes, nice catch. I will update the implementation! Thanks.